resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/SubmittedDependencyInitContainerConfigPlugin.scala

```diff
@@ -34,36 +35,62 @@ private[spark] trait SubmittedDependencyInitContainerConfigPlugin {
     jarsResourceId: String,
     filesResourceId: String,
     jarsSecretKey: String,
     filesSecretKey: String,
     trustStoreSecretKey: String,
-    secretsVolumeMountPath: String,
-    resourceStagingServiceSslOptions: SSLOptions)
+    clientCertSecretKey: String,
+    resourceStagingServerSslEnabled: Boolean,
+    maybeInternalTrustStoreUri: Option[String],
+    maybeInternalClientCertUri: Option[String],
+    maybeInternalTrustStorePassword: Option[String],
+    maybeInternalTrustStoreType: Option[String],
+    secretsVolumeMountPath: String)
     extends SubmittedDependencyInitContainerConfigPlugin {
 
   override def configurationsToFetchSubmittedDependencies(): Map[String, String] = {
     Map[String, String](
-      RESOURCE_STAGING_SERVER_URI.key -> resourceStagingServerUri,
+      RESOURCE_STAGING_SERVER_URI.key -> internalResourceStagingServerUri,
       INIT_CONTAINER_DOWNLOAD_JARS_RESOURCE_IDENTIFIER.key -> jarsResourceId,
       INIT_CONTAINER_DOWNLOAD_JARS_SECRET_LOCATION.key ->
         s"$secretsVolumeMountPath/$jarsSecretKey",
       INIT_CONTAINER_DOWNLOAD_FILES_RESOURCE_IDENTIFIER.key -> filesResourceId,
       INIT_CONTAINER_DOWNLOAD_FILES_SECRET_LOCATION.key ->
         s"$secretsVolumeMountPath/$filesSecretKey",
-      RESOURCE_STAGING_SERVER_SSL_ENABLED.key ->
-        resourceStagingServiceSslOptions.enabled.toString) ++
-      resourceStagingServiceSslOptions.trustStore.map { _ =>
-        (RESOURCE_STAGING_SERVER_TRUSTSTORE_FILE.key,
-          s"$secretsVolumeMountPath/$trustStoreSecretKey")
-      }.toMap ++
-      resourceStagingServiceSslOptions.trustStorePassword.map { password =>
+      RESOURCE_STAGING_SERVER_SSL_ENABLED.key -> resourceStagingServerSslEnabled.toString) ++
+      resolveSecretPath(
+        maybeInternalTrustStoreUri,
+        trustStoreSecretKey,
+        RESOURCE_STAGING_SERVER_TRUSTSTORE_FILE,
+        "TrustStore URI") ++
+      resolveSecretPath(
+        maybeInternalClientCertUri,
+        clientCertSecretKey,
+        RESOURCE_STAGING_SERVER_CLIENT_CERT_PEM,
+        "Client certificate URI") ++
+      maybeInternalTrustStorePassword.map { password =>
         (RESOURCE_STAGING_SERVER_TRUSTSTORE_PASSWORD.key, password)
       }.toMap ++
-      resourceStagingServiceSslOptions.trustStoreType.map { storeType =>
+      maybeInternalTrustStoreType.map { storeType =>
         (RESOURCE_STAGING_SERVER_TRUSTSTORE_TYPE.key, storeType)
       }.toMap
   }
+
+  private def resolveSecretPath(
+      maybeUri: Option[String],
+      secretKey: String,
+      configEntry: OptionalConfigEntry[String],
+      uriType: String): Map[String, String] = {
+    maybeUri.map(Utils.resolveURI).map { uri =>
+      val resolvedPath = Option(uri.getScheme).getOrElse("file") match {
+        case "file" => s"$secretsVolumeMountPath/$secretKey"
+        case "local" => uri.getPath
+        case invalid => throw new SparkException(s"$uriType has invalid scheme $invalid must be" +
```

Comment:
maybe in a fixup commit after this merges can add "has invalid scheme for secret" to give end user just a bit more context